libcxx/include/experimental/__simd/reference.h

```diff
@@ -60,6 +61,32 @@ class __simd_reference {
     return {__s_, __idx_};
   }
 
+  template <class _Up = value_type, class = decltype(std::declval<_Up>() + _Up{1})>
```

Comment:
```suggestion
  template <class _Up = value_type, class = decltype(std::declval<_Up>()++)>
```
Otherwise floating point types would have a `++`. Please add a test to make sure that's not the case.